collector/src/main/java/io/prometheus/jmx/JmxCollector.java

```diff
@@ -237,25 +237,46 @@ static String toSnakeAndLowerCase(String attrName) {
       return resultBuilder.toString();
     }
 
+  /**
+   * Change invalid chars to underscore, and merge underscores.
+   * @param name Input string
+   * @return
+   */
+  static String safeName(String name) {
+      if (name == null) {
+        return null;
+      }
+      boolean prevCharIsUnderscore = false;
+      StringBuilder safeNameBuilder = new StringBuilder();
+      for (char nameChar : name.toCharArray()) {
+        boolean isUnsafeChar = !(Character.isLetterOrDigit(nameChar) || nameChar == ':' || nameChar == '_');
+        if ((isUnsafeChar || nameChar == '_') && !prevCharIsUnderscore) {
+          safeNameBuilder.append("_");
+          prevCharIsUnderscore = true;
+        } else if (nameChar == '_' || isUnsafeChar) {
```

Comment:
I think this would be clearer as an if inside the previous if

Reply:
Thanks! That simplified the logic substantially.